docs/development/compatibility-policy.rst

```diff
@@ -298,99 +303,105 @@ When an application wants to be upgraded to a new version of Twisted, it can do
 However, if the application wants to get the same **for free** behavior for the next upgrade, the application's tests should be run treating warnings as errors, and fixed.
 
 
-Supporting and de-supporting Python versions
+Supporting and De-supporting Python Versions
 --------------------------------------------
 
 Twisted does not have a formal policy around supporting new versions of Python or de-supporting old versions of Python.
 We strive to support Twisted on any version of Python that is the default Python for a vendor-supported release from a major platform, namely Debian, Ubuntu, the latest release of Windows, or the latest release of macOS.
-The versions of Python currently supported are listed in the ​INSTALL file for each release.
 
-A distribution release + Python version is only considered supported when a `buildbot builder <http://buildbot.twistedmatrix.com>`_ exists for it.
+A distribution release + Python version is only considered supported when a `GitHub Actions test workflow <https://github.com/twisted/twisted/blob/trunk/.github/workflows/test.yaml>`_ exists for it.
 
 Removing support for a Python version will be announced at least 1 release prior to the removal.
 
 
-How to deprecate APIs
+How to Deprecate APIs
 ---------------------
 
 
 Classes
 ^^^^^^^
 
-Classes are deprecated by raising a warning when they are access from within their module, using the :py:func:`deprecatedModuleAttribute <twisted.python.deprecate.deprecatedModuleAttribute>` helper.
+Deprecate a class by raising a warning when it is accessed within its module,
+using the :py:func:`deprecatedModuleAttribute <twisted.python.deprecate.deprecatedModuleAttribute>` helper after the class definition:
 
 .. code-block:: python
 
     class SSLContextFactory:
         """
         An SSL context factory.
         """
-        deprecatedModuleAttribute(
-            Version("Twisted", 12, 2, 0),
-            "Use twisted.internet.ssl.DefaultOpenSSLContextFactory instead.",
-            "twisted.mail.protocols", "SSLContextFactory")
 
+    deprecatedModuleAttribute(
+        Version("Twisted", "NEXT", 0, 0),
+        "Use twisted.internet.ssl.DefaultOpenSSLContextFactory instead.",
+        __name__,
+        SSLContextFactory.__name__,
+    )
 
-Functions and methods
-^^^^^^^^^^^^^^^^^^^^^
+Pass ``Version("Twisted", "NEXT", 0, 0)`` `incremental placeholder <https://github.com/twisted/incremental#updating>`_ to the to indicate the upcoming release.
```

Comment:
Shouldn't we recommand a single placeholder ?

I would go with ``Twisted NEXT`` as it should be valid for both Python or RST code.

Reply:
They're not interchangeable: `Version(...)` is what you pass to the APIs; `Twisted NEXT` is what you use in docstrings and/or assertions.

Reply:
Thanks for the info.
I see why you might want to have an increment Version instance in Python... but I always felt that this is an overengineering ... at least for the deprecation warning message.